### ISSUE TITLE
Platform code is not included in Maui class library project

### ISSUE THREAD

**Issue by JPfahl** [dotnet/maui#9079]
### Description

I am running 6.0.419 - the dropdowns below need updating. 

I have moved code into a class library that I include in my project and have found that code in platforms is not included when the library is compiled.  This worked when it was in the application project.

My files are structured this way:

![image](https://user-images.githubusercontent.com/101600436/181817189-3d8a6a62-aed4-46ab-8ddc-5c8f780e3328.png)

And the code uses a static constructor to run the mapper:

using Microsoft.Maui.Handlers;

namespace Computronix.Maui.Widgets;

public partial class EntryBase : Entry
{
    static EntryBase()
    {
        EntryHandler.Mapper.AppendToMapping(
            "EntryBase",
            (handler, view) =>
            {
                handler.PlatformView.BorderStyle = UIKit.UITextBorderStyle.None;
            });
    }
}

I have included the following in the prog file:

	<ItemGroup Condition="$(TargetFramework.StartsWith('net6.0-android')) != true">
		<Compile Remove="**\**\*.Android.cs" />
		<None Include="**\**\*.Android.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
		<Compile Remove="**\Android\**\*.cs" />
		<None Include="**\Android\**\*.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
	</ItemGroup>

	<ItemGroup Condition="$(TargetFramework.StartsWith('net6.0-ios')) != true AND $(TargetFramework.StartsWith('net6.0-maccatalyst')) != true">
		<Compile Remove="**\**\*.iOS.cs" />
		<None Include="**\**\*.iOS.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
		<Compile Remove="**\iOS\**\*.cs" />
		<None Include="**\iOS\**\*.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
	</ItemGroup>

	<ItemGroup Condition="$(TargetFramework.Contains('-windows')) != true ">
		<Compile Remove="**\*.Windows.cs" />
		<None Include="**\*.Windows.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
		<Compile Remove="**\Windows\**\*.cs" />
		<None Include="**\Windows\**\*.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
	</ItemGroup>

### Steps to Reproduce

Compile library
Include in Application
run app
constructor is not called.

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows

### Affected platform versions

iOS 15.5

### Did you find any workaround?

using compiler directives works.

Not near as clean as using the platforms structure.

#if IOS
        EntryHandler.Mapper.AppendToMapping(
            "EntryBase",
            (handler, view) =>
            {
                handler.PlatformView.BorderStyle = UIKit.UITextBorderStyle.None;
            });
#endif

**Comment by Eilon** [comment]
Is your class library a .NET MAUI Class Library, or a "regular" class library? Only MAUI class libraries have this "platforms" feature automatically.

**Comment by JPfahl** [comment]
I built a new maui class library in preview 5 and moved the code over.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Eilon** [comment]
Alright thanks. And just checking: can you share why the static constructor _should_ have been called? Was the `EntryBase` type being used anywhere in the app? The static constructors are only guaranteed to be called prior to the first use of the type in an app. If the app doesn't use the type, the constructor won't ever get called at all.

**Comment by JPfahl** [comment]
Yes, I’m using the class on a page.  The entry renders but the border is not removed I placed a debug.writeline in the constructor and it newer gets executed.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Eilon** [comment]
OK that definitely seems odd, then!

**Comment by JkPersson** [comment]
I am seeing a related issue. I created a class Maui class library and created a partial class with implementation in the platform code.

Non-platform:
`public partial class BluetoothConnection : ObservableObject
{
    public partial void Init(string macAddress, BluetoothParser parser);
}
`

Platform code:
`public partial class BluetoothConnection
{
    public partial void Init(string macAddress, BluetoothParser parser){...}
}
`

Despite being able to navigate between them using the Visual Studio references feature, I get the following compiler error:

_Partial method 'BluetoothConnection.Init(string, BluetoothParser)' must have an implementation part because it has accessibility modifiers.	Brunata.Handterminal.Visit.Shared (net6.0)_

**Comment by jfversluis** [comment]
I've seen a couple of this issues and usually it comes down to something not being right in there. It would be very helpful if someone could add a reproduction project in the form of a public GitHub repo that we can look at and hopefully easily point out what isn't looking right.
